qek/data/extractors.py

```diff
@@ -351,65 +351,79 @@ def __init__(
 
         super().__init__(path=path, device=device, compiler=compiler)
         self._sdk = sdk
-        self._batch_id = batch_id
+        self._batch_ids: list[str] | None = batch_ids
 
     @property
     def batch_ids(self) -> list[str] | None:
-        return self._batch_id
+        return self._batch_ids
 
     async def run(self) -> list[ProcessedData]:
         if len(self.sequences) == 0:
             logger.warning("No sequences to run, did you forget to call compile()?")
             return []
 
-        batches: list[Batch] = []
-        if self._batch_id is None:
+        device: pl.devices.Device = self.sequences[0].sequence.device
+        # The API doesn't support run longer than 500 jobs.
+        # If we want to add more runs, we'll need to split them across several jobs.
+        max_runs = device.max_runs if isinstance(device.max_runs, int) else 500
+
+        if self._batch_ids is None:
             # Enqueue jobs.
             self._batch_ids = []
             for compiled in self.sequences:
-                logger.debug("Executing compiled graph #%s", id)
+                logger.debug("Enqueuing execution of compiled graph #%s", compiled.graph.id)
```

Comment:
```suggestion
                logger.debug(f"Enqueuing execution of compiled graph {compiled.graph.id}")
```

Reply:
In most languages/frameworks that I know of, it's considered bad form to do that, because it forces you to compute the debug string (which could be expensive) even if you deactivate logging.

I know that Python holds a strong suspicion against anything that looks remotely like performance, but I kinda assume that it's the same in that language :)